ludwig/schema/hyperopt/scheduler.py

```diff
@@ -0,0 +1,641 @@
+            " each trial as it finishes one step of training. The callable must take four arguments: TrialRunner, "
+            "current Trial, current result dict and the ResourceChangingScheduler calling it. The callable must return "
+            "a PlacementGroupFactory, Resources, dict or None (signifying no need for an update). If "
+            "resources_allocation_function is None, no resource requirements will be changed at any time. By default, "
+            "DistributeResources will be used, distributing available CPUs and GPUs over all running trials in a robust"
+            " way, without any prioritization."
+        )
+    )
+
+
+def get_scheduler_conds():
+    """Returns a JSON schema of conditionals to validate against optimizer types defined in
+    `ludwig.modules.optimization_modules.optimizer_registry`."""
+    conds = []
+    for scheduler_config in scheduler_config_registry:
+        scheduler_cls = scheduler_config_registry[scheduler_config]
+        other_props = schema_utils.unload_jsonschema_from_marshmallow_class(scheduler_cls)["properties"]
+        schema_utils.remove_duplicate_fields(other_props)
+        preproc_cond = schema_utils.create_cond(
+            {"type": scheduler_config},
+            other_props,
+        )
+        conds.append(preproc_cond)
+    return conds
+
+
+def SchedulerDataclassField(default={"type": "async_hyperband"}, description="Hyperopt scheduler settings."):
+    """Custom dataclass field that when used inside of a dataclass will allow any scheduler in
+    `ludwig.schema.hyperopt.scheduler.scheduler_registry`.
+
+    Sets default optimizer to 'async_hyperband'.
+
+    :param default: Dict specifying an optimizer with a `type` field and its associated parameters. Will attempt to use
+           `type` to load scheduler from registry with given params. (default: {"type": "async_hyperband"}).
+    :return: Initialized dataclass field that converts untyped dicts with params to optimizer dataclass instances.
+    """
+
+    class OptimizerMarshmallowField(fields.Field):
+        """Custom marshmallow field that deserializes a dict to a valid optimizer from
+        `ludwig.modules.optimization_modules.optimizer_registry` and creates a corresponding `oneOf` JSON schema
+        for external usage."""
+
+        def _deserialize(self, value, attr, data, **kwargs):
+            if value is None:
+                return None
+            if isinstance(value, dict):
+                if "type" in value and value["type"] in scheduler_config_registry:
+                    opt = scheduler_config_registry[value["type"].lower()][1]
+                    try:
+                        return opt.Schema().load(value)
+                    except (TypeError, ValidationError) as e:
+                        raise ValidationError(
+                            f"Invalid params for optimizer: {value}, see `{opt}` definition. Error: {e}"
```

Comment:
Should be scheduler instead of optimizer

Reply:
will fix.